google/cloud/pubsub/internal/tracing_message_batch.cc

```diff
@@ -43,6 +44,11 @@ using Attributes =
 using Links =
     std::vector<std::pair<opentelemetry::trace::SpanContext, Attributes>>;
 
+int64_t GetCurrentThreadId() {
```

Comment:
Refactor and use this?

https://github.com/googleapis/google-cloud-cpp/blob/45b5c659b146729b1bb85ad2f64d2218265a7b8c/google/cloud/storage/internal/async/writer_connection_tracing.cc#L33-L37

Reply:
The otel recc says that the attribute should be an [int](https://opentelemetry.io/docs/specs/semconv/general/attributes/#general-thread-attributes). I think the other spot should be refactored. 

By refactor, do you mean make this a common function in opentelemetry.h?

Reply:
I wonder how they expect to get an `int` out of [std::thread::id](https://en.cppreference.com/w/cpp/thread/thread/id).  There is no guaranteed conversion to any integral type.  Just formatting to `std::iostream`.

Reply:
https://github.com/open-telemetry/semantic-conventions/issues/546